app/controllers/owners_controller.rb

```diff
@@ -0,0 +1,56 @@
+class OwnersController < ApplicationController
+  before_action :find_rubygem, except: :confirm
+  before_action :redirect_to_signin, unless: :owner?, except: %i[confirm resend_confirmation]
+
+  def confirm
+    ownership = Ownership.includes(:rubygem).find_by!(token: params[:token])
+
+    if ownership.valid_confirmation_token?
+      ownership.confirm_and_notify
+      redirect_to rubygem_path(ownership.rubygem), notice: t(".confirm.confirmed_email", gem: ownership.rubygem.name)
+    else
+      redirect_to root_path, alert: t(".confirm.token_expired")
+    end
+  end
+
+  def resend_confirmation
+    ownership = @rubygem.ownerships_including_unconfirmed.find_by!(user: current_user)
+    ownership.generate_confirmation_token && ownership.save
+    OwnersMailer.delay.ownership_confirmation(ownership.id)
+    redirect_to rubygem_path(ownership.rubygem), notice: t("owners.resend_confirmation.resent_notice", handle: ownership.user.handle)
+  end
+
+  def index
+    @ownerships = @rubygem.ownerships_including_unconfirmed.includes(:user, :authorizer)
+  end
+
+  def create
+    owner = User.find_by_name(params[:owner])
```

Comment:
I can add a new method to `user.rb` for this. But isn't an alert with _Owner could not be found._ more intuitive to user than a 404 page?
We are following the same in API controller too.

Reply:
It would nicer if we can show error message but `if owner` is a bit awkward. Perhaps we can handle this with validation of `ownership`? ie if `user` is `nil` show appropriate flash/error.